react/index.tsx

```diff
@@ -1,88 +1,69 @@
-import React, { Fragment, FC, useRef, useEffect, useCallback } from 'react'
+import React, { Fragment, FC, useEffect } from 'react'
 import { path } from 'ramda'
 import { useCssHandles } from 'vtex.css-handles'
+import { useStickyScroll } from './useStickyScroll'
 
 const CSS_HANDLES = ['container'] as const
 
 enum Positions {
   BOTTOM = 'bottom',
+  TOP = 'top',
 }
 
 interface Props {
   position?: Positions
+  verticalSpacing?: number
 }
 
 interface StorefrontComponent extends FC<Props> {
   schema?: any
 }
 
-const StickyLayoutComponent: StorefrontComponent = ({
-  children,
-  position,
-}) => {
-  const container = useRef<HTMLDivElement>(null)
+const StickyLayoutComponent: StorefrontComponent = ({ children, position, verticalSpacing = 0 }) => {
   const handles = useCssHandles(CSS_HANDLES)
-
-  if (position !== Positions.BOTTOM) {
-    // Only 'bottom' position supported for now!
-    return <Fragment>{children}</Fragment>
-  }
-
-  const handleScroll = useCallback((e?: Event) => {
-    const target = (e && e.target) as HTMLDivElement | null
-    if (!container.current) {
-      return
-    }
-
-    // We have to get this baseTop variable to handle cases when a modal changes the top style of the base element, resetting the window.pageYOfsset
-    const baseTop = parseInt(
-      path(['scrollingElement', 'style', 'top'], target) || '0',
-      10
-    )
-    const componentTop = container.current.offsetTop + baseTop
-
-    const newTop =
-      window.innerHeight -
-      componentTop -
-      container.current.clientHeight +
-      window.pageYOffset
-
-    if (newTop < 0) {
-      container.current.style.transform = `translate3d(0, ${newTop}px, 0)`
-      return
-    }
-    if (newTop >= 0) {
-      // Reset and don't change anything
-      container.current.style.transform = `translate3d(0, 0px, 0)`
-    }
-  }, [])
+  const {
+    containerReference,
+    container,
+    handlePosition,
+    debouncedHandlePosition
```

Comment:
⛏ expose only `debouncedHandlePosition`, without the `debounced` part, and use it on both handlers

Reply:
Done.